flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CodeAnalyzer.java

```diff
@@ -396,17 +398,32 @@ public void visit(VariableDeclarationNode variableDeclarationNode) {
 
         // Generate the default expression node if a node is not built
         if (isNodeUnidentified()) {
-            startNode(FlowNode.Kind.NEW_DATA)
-                    .metadata()
+            if (initializerNode.kind() == SyntaxKind.BRACED_EXPRESSION) {
+                initializerNode = CommonUtils.getExpressionNodeFromBracedExpression(
+                        (BracedExpressionNode) initializerNode);
+            }
+            if (initializerNode.kind() == SyntaxKind.XML_TEMPLATE_EXPRESSION) {
+                startNode(FlowNode.Kind.NEW_XML_PAYLOAD_DATA)
+                        .metadata()
+                        .description(NewXMLPayloadData.DESCRIPTION)
+                        .stepOut()
+                        .properties().expression(initializerNode);
+            } else {
+                startNode(FlowNode.Kind.NEW_DATA)
+                        .metadata()
```

Comment:
We can replace all these lines with a single visit method:

```java
@Override
public void visit(TemplateExpressionNode templateExpressionNode) {
    if (templateExpressionNode.kind() == SyntaxKind.XML_TEMPLATE_EXPRESSION) {
        startNode(FlowNode.Kind.NEW_XML_PAYLOAD_DATA)
                .metadata()
                .description(NewXMLPayloadData.DESCRIPTION)
                .stepOut()
                .properties().expression(templateExpressionNode);
    }
}
```

We also need to reevaluate whether we can use `new xml payload`. With the new proposed design, we are prioritizing the expression, and let the user decide how the variable assignment should be.

Reply:
As I understand the requirement new xml payload used to represent variable declarations with xml template literal values in L.H.S. If we move the visitor to the template expression node level, when we have xml template literal any place we will represent it as a new xml payload. Is that our requirement? @hasithaa

Reply:
updated